python/ray/data/datasource/lance_datasource.py

```diff
@@ -73,11 +73,12 @@ def estimate_inmemory_data_size(self) -> Optional[int]:
         return None
 
 
-def _read_fragments(fragments, columns, row_filter) -> Iterator["pyarrow.Table"]:
+def _read_fragments(fragment_ids, lance_ds, columns, row_filter) -> Iterator["pyarrow.Table"]:
     """Read Lance fragments in batches."""
     import pyarrow
 
-    for fragment in fragments:
+    for fragment_id in fragment_ids:
+        fragment = lance_ds.get_fragment(fragment_id)
         batches = fragment.to_batches(columns=columns, filter=row_filter)
         for batch in batches:
             yield pyarrow.Table.from_batches([batch])
```

Comment:
If you wanted something that did some IO prefetching, you could instead do:

```suggestion
    fragments = [lance_ds.get_fragment(id) for id in fragment_ids]
    scanner = lance_ds.scanner(
        columns,
        filter=row_filter,
        fragments=fragments,
    )
    for batch in scanner.to_reader():
        yield pyarrow.Table.from_batches([batch])
```